core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java

```diff
@@ -66,180 +94,131 @@ public static RegisteredType of(CatalogItem<?, ?> item) {
-        
-        @Override
-        public boolean isDeprecated() {
-            return deprecated;
-        }
-        
-        @Override
-        public Class<?> getJavaType() {
-            return javaType;
-        }
-        
-        @Override
-        public String toString() {
-            return JavaClassNames.simpleClassName(this)+"["+getId()+
-                (isDisabled() ? ";DISABLED" : "")+
-                (isDeprecated() ? ";deprecated" : "")+
-                "]";
-        }
+    /** Returns a wrapped map, if the object is YAML which parses as a map; 
+     * otherwise returns absent capable of throwing an error with more details */
+    @SuppressWarnings("unchecked")
+    public static Maybe<Map<Object,Object>> getAsYamlMap(Object planData) {
+        if (!(planData instanceof String)) return Maybe.absent("not a string");
+        Iterable<Object> result;
+        try {
+            result = Yamls.parseAll((String)planData);
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            return Maybe.absent(e);
+        }
+        Iterator<Object> ri = result.iterator();
+        if (!ri.hasNext()) return Maybe.absent("YAML has no elements in it");
+        Object r1 = ri.next();
+        if (ri.hasNext()) return Maybe.absent("YAML has multiple elements in it");
+        if (r1 instanceof Map) return Maybe.of((Map<Object,Object>)r1);
+        return Maybe.absent("YAML does not contain a map");
     }
 
-    // TODO
-//    public static class RegisteredBeanType extends AbstractRegisteredType {
-//        
-//    }
+    /** 
+     * Queries recursively the supertypes of {@link RegisteredType} to see whether it 
+     * declares a supertype compatible with the given {@link Class} */
+    public static boolean isSubTypeOf(RegisteredType type, Class<?> superType) {
+        return isSubTypeOf(type.getSuperTypes(), superType);
+    }
     
-    public static class RegisteredSpecType extends AbstractRegisteredType {
-
-        private TypeImplementation impl;
-        
-        public RegisteredSpecType(String symbolicName, String version, Class<?> javaType, TypeImplementation impl) {
-            super(symbolicName, version, javaType);
-            this.impl = impl;
+    /** 
+     * Queries recursively the given types (either {@link Class} or {@link RegisteredType}) 
+     * to see whether any are compatible with the given {@link Class} */
+    public static boolean isSubTypeOf(Set<Object> allKnownTypes, Class<?> superType) {
+        for (Object st: allKnownTypes) {
+            if (st instanceof Class) {
+                if (superType.isAssignableFrom((Class<?>)st)) return true;
+            }
         }
-
-        public TypeImplementation getImplementation() {
-            return impl;
+        for (Object st: allKnownTypes) {
+            if (st instanceof RegisteredType) {
+                if (isSubTypeOf((RegisteredType)st, superType)) return true;
+            }
         }
+        return false;
```

Comment:
Looks the same as `RegisteredTypePredicates.assignableFrom(Class<?>)`? Re-use implementation and unify names.